libsupport/include/katana/ArrowInterchange.h

```diff
@@ -276,6 +276,110 @@ TableBuilder::AddColumn(const ColumnOptions& options) {
   columns_.emplace_back(std::make_shared<arrow::ChunkedArray>(chunks));
 }
 
+////////////////////////////////////////////
+// Arrow utilities
+
+KATANA_EXPORT std::shared_ptr<arrow::Array> Unchunk(
+    const std::shared_ptr<arrow::ChunkedArray>& original);
+KATANA_EXPORT std::shared_ptr<arrow::ChunkedArray> Shuffle(
+    const std::shared_ptr<arrow::ChunkedArray>& original);
+
+template <typename ArrowArrayType>
+void
+PrintFirstNonzeroElts(
+    const std::string& header,
+    const std::shared_ptr<arrow::ChunkedArray>& chunked_array,
+    int32_t num_elts) {
+  auto cast_res = ViewCast<ArrowArrayType>(Unchunk(chunked_array));
+  if (!cast_res) {
+    KATANA_LOG_DEBUG("failed to cast array: {}", cast_res.error());
+    return;
+  }
+  auto base = cast_res.value();
+  for (int64_t i = 0; i < base->length(); ++i) {
+    if (base->Value(i) != 0) {
+      std::string tmp = header;
+      tmp.append(":");
+      int64_t end = std::min(i + num_elts, base->length());
+      for (int64_t j = i; j < end; j += 3) {
+        switch (end - j) {
+        case 1: {
+          fmt::print("{:8} {:7}: {:8}\n", tmp, j, base->Value(j));
+          break;
+        }
+        case 2: {
+          fmt::print(
+              "{:8} {:7}: {:8} {:7}: {:8}\n", tmp, j, base->Value(j), j + 1,
+              base->Value(j + 1));
+          break;
+        }
+        default: {
+          fmt::print(
+              "{:8} {:7}: {:8} {:7}: {:8} {:7}: {:8}\n", tmp, j, base->Value(j),
+              j + 1, base->Value(j + 1), j + 2, base->Value(j + 2));
+          break;
+        }
+        }
+      }
+      return;
+    }
+  }
+}
+
+template <typename ArrowArrayType>
+void
+PrintCmpElts(
```

Comment:
"Elements"

"Cmp"?

Given the rather precise formatting used in this function, it is hard for me to see how this function or the above would be generally reusable. Would it be possible to just keep this in the utility and wait to see if someone else really wants to share/reuse this?

Reply:
Sure thing.  I realize it was a stretch, but these functions answer the "what is in this array" and "when these two arrays aren't equal, why not?" questions.  That came up multiple times in multiple files for me, but I'll take my snacks out of the break room.